commit 78055361d20ca94e6c59e032e3a6ade420affb91
Author: Eike Hein <hein@kde.org>
Date:   Sun Sep 29 10:49:00 2013 +0200

    Fix rare out-of-bounds list indexing when inserting launchers.
    
    BUG:324218

diff --git a/libs/taskmanager/taskgroup.cpp b/libs/taskmanager/taskgroup.cpp
index 8df5c9a..d174fcb 100644
--- a/libs/taskmanager/taskgroup.cpp
+++ b/libs/taskmanager/taskgroup.cpp
@@ -210,11 +210,11 @@ void TaskGroup::add(AbstractGroupableItem *item, int insertIndex)
         if (d->groupManager->separateLaunchers()) {
             if (item->itemType() == LauncherItemType) {
                 KUrl lUrl = item->launcherUrl();
-                int urlIdx = d->groupManager->launcherIndex(lUrl);
 
-                int maxIndex = urlIdx;
-                if(maxIndex < 0) {
-                    maxIndex = d->members.count();
+                int maxIndex = d->groupManager->launcherIndex(lUrl);
+
+                if (maxIndex < 0 || maxIndex >= d->members.count()) {
+                    maxIndex = d->members.count() - 1;
                 }
 
                 for (index = 0; index < maxIndex; ++index) {
